src/app/dashboard/whitelist/WhiteListPage.js

```diff
@@ -0,0 +1,392 @@
+// @flow
+
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import DocumentTitle from 'react-document-title'
+import uuidv4 from 'uuid/v4'
+import {
+  DataTable,
+  PaginationV2,
+  Modal,
+  ModalWrapper,
+  DatePicker,
+  DatePickerInput,
+  FileUploaderButton,
+} from 'carbon-components-react'
+import { TransferManager } from 'polymathjs'
+
+import type { Investor } from 'polymathjs/types'
+
+import { initialize, uploadCSV, multiUserSubmit, oneUserSubmit, getWhitelist, listLength, removeInvestor, editInvestors } from './actions'
+import { TableHeaders } from './tableHeaders'
+import InvestorForm from './components/userForm'
+import EditInvestorsForm from './components/editInvestorsForm'
+import BasicDropzone from './components/ReactDropZone'
+
+import './style.css'
+
+// might need TableToolbarAction and batchActionClick here
+const { Table, TableBody, TableCell, TableContainer, TableHead, TableHeader, TableRow,
+  TableSelectAll, TableSelectRow, TableToolbar, TableBatchAction, TableBatchActions,
+  TableToolbarSearch, TableToolbarContent } = DataTable
+
+type StateProps = {|
+  transferManager: TransferManager,
+  addresses: Array<string>,
+  sell: Array<number>,
+  buy: Array<number>,
+  investors: Array<Investor>,
+  csvMessage: string,
+  previewCSVShowing: boolean,
+  listLength: number,
+|}
+
+type DispatchProps = {|
+  initialize: () => any,
+  handleUpload: () => any,
+  multiSubmit: () => any,
+  singleSubmit: () => any,
+  getWhitelist: (?Date, ?Date) => any,
+  updateListLength: (number) => any,
+  removeInvestor: (investors: Array<string>) => any,
+  editInvestors: (investors: Array<string>) => any,
+|}
+
+const mapStateToProps = (state) => ({
+  transferManager: state.whitelist.transferManager,
+  addresses: state.whitelist.addresses,
+  sell: state.whitelist.sell,
+  buy: state.whitelist.buy,
+  investors: state.whitelist.investors,
+  csvMessage: state.whitelist.csvMessage,
+  previewCSVShowing: state.whitelist.previewCSVShowing,
+  listLength: state.whitelist.listLength,
```

Comment:
seems that you can just import entire whitelist state here via only one string

Reply:
yes, i updated it to look like this. it is much cleaner looking now